File: additional/internal/services/cmd/rotate/rotate.go
// Пакет для обработки команды поворота
package rotate

import (
	"additional/internal/services/rotating"
)

type Rotate struct {
	object rotating.IRotable
}

func NewRotate(object rotating.IRotable) *Rotate {
	return &Rotate{object: object}
}

func (r *Rotate) Execute() error {
	angle, err := r.object.GetAngle()
	if err != nil {
		return err
	}

	angularVelocity, err := r.object.GetAngularVelocity()
	if err != nil {
		return err
	}

	division, err := r.object.GetDivision()
	if err != nil {
		return err
	}

	newValue := (angle + angularVelocity) % division
	return r.object.SetAngle(newValue)
}

File: main/src/server/cmd/rotate.py
import zope

from server.errors import RotateParamException
from server.interfaces.cmd import ICommand
from server.models import Angle


@zope.interface.implementer(ICommand)
class RotateCommand:
    """Команда поворота объекта"""

    def __init__(self, object):
        self.object = object

    def execute(self) -> None:
        """Установка нового угла при повороте"""

        alpha = self.object.get_angle()
        if not isinstance(alpha, Angle):
            raise RotateParamException('Incorrect alpha value')

        angular_velocity = self.object.get_angular_velocity()
        if not isinstance(angular_velocity, int):
            raise RotateParamException('Incorrect angular velocity value')

        division = self.object.get_division()
        new_angle = (alpha.value + angular_velocity) % division
        self.object.set_angle(Angle(new_angle, division))
